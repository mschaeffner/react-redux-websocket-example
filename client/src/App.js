import React from 'react'
import { connect } from 'react-redux'
import LoginScreen from './LoginScreen'
import ChatScreen from './ChatScreen'


const mapStateToProps = state => ({
  connectionToServer: state.connectionToServer
})

const App = ({connectionToServer}) =>
  (connectionToServer) ? <ChatScreen /> : <LoginScreen /> 

export default connect(mapStateToProps)(App)
