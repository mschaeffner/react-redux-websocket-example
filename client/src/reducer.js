import {
  CHANGE_NEW_MESSAGE,
  SEND_NEW_MESSAGE,
  RECEIVE_MESSAGE
} from './actions'


const INITIAL_STATE = {
  newMessageText: '',
  messages: [{sender:"acsc", text:"aascsac"}]
}

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {

    case CHANGE_NEW_MESSAGE:
      return { ...state, newMessageText: action.payload}

    case SEND_NEW_MESSAGE:
      return { ...state, newMessageText: ''}

    case RECEIVE_MESSAGE:
      const newMessage = action.payload
      const messages = [ ...state.messages, newMessage]
      return { ...state, messages}

    default:
      return state
  }
}

