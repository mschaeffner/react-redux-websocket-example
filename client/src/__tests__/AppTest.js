import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import reducer from '../reducer.js'
import App from '../App'


const STATE_ONLY_ME_LOGGED_IN = {
  activeUsers: [
    {id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias", messages: []}
  ],
  connectionToServer: "Websocket",
  me: { id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias" },
  messages: [],
  newMessageText: "",
  selectedChannel: "ALL"
}


const STATE_WITH_3_USERS_AND_SOME_PUBLIC_MESSAGES = {
  activeUsers: [
    {id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias", messages: []},
    {id: "1d14f01e-69d9-4de9-9996-869fc291fbf8", username: "alice", messages: []},
    {id: "2d14f01e-69d9-4de9-9996-869fc291fbf8", username: "bob", messages: []},
  ],
  connectionToServer: "Websocket",
  me: { id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias" },
  messages: [
    {
      receiver: "ALL",
      sender: { id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias" },
      text: "the first message"
    },
    {
      receiver: "ALL",
      sender: { id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias" },
      text: "the second message"
    },
    {
      receiver: "ALL",
      sender: { id: "1d14f01e-69d9-4de9-9996-869fc291fbf8", username: "alice" },
      text: "a message from alice"
    },
    {
      receiver: "ALL",
      sender: { id: "2d14f01e-69d9-4de9-9996-869fc291fbf8", username: "bob" },
      text: "a message from bob"
    }
  ],
  newMessageText: "starting of a new message",
  selectedChannel: "ALL"
}


const STATE_WITH_3_USERS_AND_SOME_PRIVATE_MESSAGES = {
  activeUsers: [
    {id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias", messages: []},
    {id: "1d14f01e-69d9-4de9-9996-869fc291fbf8", username: "alice", messages: [
      {
        receiver: "1d14f01e-69d9-4de9-9996-869fc291fbf8",
        sender: { id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias" },
        text: "a private message to alice"
      },
      {
        receiver: "0d14f01e-69d9-4de9-9996-869fc291fbf8",
        sender: { id: "1d14f01e-69d9-4de9-9996-869fc291fbf8", username: "alice" },
        text: "a private message from alice back to mathias"
      }
    ]},
    {id: "2d14f01e-69d9-4de9-9996-869fc291fbf8", username: "bob", messages: []},
  ],
  connectionToServer: "Websocket",
  me: { id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias" },
  messages: [
    {
      receiver: "ALL",
      sender: { id: "0d14f01e-69d9-4de9-9996-869fc291fbf8", username: "mathias" },
      text: "the first message"
    },
    {
      receiver: "ALL",
      sender: { id: "1d14f01e-69d9-4de9-9996-869fc291fbf8", username: "alice" },
      text: "the second message"
    }
  ],
  newMessageText: "starting of a new message",
  selectedChannel: "1d14f01e-69d9-4de9-9996-869fc291fbf8"
}



describe('App', () => {


  it('should render the login screen', () => {
    const store = createStore(reducer)
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })


  it('should render the app with only me logged in', () => {
    const store = createStore(reducer, STATE_ONLY_ME_LOGGED_IN)
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })


  it('should render the app with 3 users logged in and some public messages', () => {
    const store = createStore(reducer, STATE_WITH_3_USERS_AND_SOME_PUBLIC_MESSAGES)
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })


  it('should render the app with 3 users logged in and some private messages', () => {
    const store = createStore(reducer, STATE_WITH_3_USERS_AND_SOME_PRIVATE_MESSAGES)
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })


})
