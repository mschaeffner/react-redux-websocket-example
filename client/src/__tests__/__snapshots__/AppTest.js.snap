// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the app with 3 users logged in and some private messages 1`] = `
<div
  className="sc-ifAKCX lnvtse"
>
  <div
    className="sc-bZQynM bJffvh"
  >
    <div
      className="sc-htpNat gdoqOL"
    >
      <div
        className=""
        onClick={[Function]}
      >
        ALL
        
      </div>
      <div
        className=""
        onClick={[Function]}
      >
        mathias
         (you)
      </div>
      <div
        className="highlight"
        onClick={[Function]}
      >
        alice
        
      </div>
      <div
        className=""
        onClick={[Function]}
      >
        bob
        
      </div>
    </div>
  </div>
  <div
    className="sc-EHOje lcMTcm"
  >
    <div
      className="sc-gzVnrw iiLelF"
    >
      <div>
        <div>
          <b>
            mathias
            :
          </b>
           
          a private message to alice
        </div>
        <div>
          <b>
            alice
            :
          </b>
           
          a private message from alice back to mathias
        </div>
      </div>
    </div>
    <div
      className="sc-htoDjs ihRtnX"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="sc-bxivhb ktBgBC"
          onChange={[Function]}
          placeholder="Type new message and hit enter"
          value="starting of a new message"
        />
      </form>
    </div>
  </div>
</div>
`;

exports[`App should render the app with 3 users logged in and some public messages 1`] = `
<div
  className="sc-ifAKCX lnvtse"
>
  <div
    className="sc-bZQynM bJffvh"
  >
    <div
      className="sc-htpNat gdoqOL"
    >
      <div
        className="highlight"
        onClick={[Function]}
      >
        ALL
        
      </div>
      <div
        className=""
        onClick={[Function]}
      >
        mathias
         (you)
      </div>
      <div
        className=""
        onClick={[Function]}
      >
        alice
        
      </div>
      <div
        className=""
        onClick={[Function]}
      >
        bob
        
      </div>
    </div>
  </div>
  <div
    className="sc-EHOje lcMTcm"
  >
    <div
      className="sc-gzVnrw iiLelF"
    >
      <div>
        <div>
          <b>
            mathias
            :
          </b>
           
          the first message
        </div>
        <div>
          <b>
            mathias
            :
          </b>
           
          the second message
        </div>
        <div>
          <b>
            alice
            :
          </b>
           
          a message from alice
        </div>
        <div>
          <b>
            bob
            :
          </b>
           
          a message from bob
        </div>
      </div>
    </div>
    <div
      className="sc-htoDjs ihRtnX"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="sc-bxivhb ktBgBC"
          onChange={[Function]}
          placeholder="Type new message and hit enter"
          value="starting of a new message"
        />
      </form>
    </div>
  </div>
</div>
`;

exports[`App should render the app with only me logged in 1`] = `
<div
  className="sc-ifAKCX lnvtse"
>
  <div
    className="sc-bZQynM bJffvh"
  >
    <div
      className="sc-htpNat gdoqOL"
    >
      <div
        className="highlight"
        onClick={[Function]}
      >
        ALL
        
      </div>
      <div
        className=""
        onClick={[Function]}
      >
        mathias
         (you)
      </div>
    </div>
  </div>
  <div
    className="sc-EHOje lcMTcm"
  >
    <div
      className="sc-gzVnrw iiLelF"
    >
      <div />
    </div>
    <div
      className="sc-htoDjs ihRtnX"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="sc-bxivhb ktBgBC"
          onChange={[Function]}
          placeholder="Type new message and hit enter"
          value=""
        />
      </form>
    </div>
  </div>
</div>
`;

exports[`App should render the login screen 1`] = `
<div
  className="sc-bwzfXH jcRXCr"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="sc-bdVaJa faSDVW"
      onChange={[Function]}
      placeholder="Choose a username and hit enter"
      value={undefined}
    />
  </form>
</div>
`;
